src/main/kotlin/com/openlattice/authorization/mapstores/PrincipalTreesMapstore.kt

```diff
@@ -125,9 +125,10 @@ class PrincipalTreesMapstore(val hds: HikariDataSource) : TestableSelfRegisterin
                     ResultSetAdapters.aclKey(it) to ResultSetAdapters.principalOfAclKey(it)
                 }
         )
-        val map: MutableMap<AclKey, AclKeySet> = mutableMapOf()
+        val map = mutableMapOf<AclKey, AclKeySet>()
         data.forEach { map.getOrPut(it.first) { AclKeySet() }.add(it.second) }
-        (map.keys - keys).forEach { map.putIfAbsent(it, AclKeySet()) }
+        ( keys - map.keys ).forEach { map[it] = AclKeySet() }
```

Comment:
this seems very important :D

Reply:
I know... this bug has been here for a while.